### PR TITLE
Adjust "abort warn others" dialog (EXPOSUREAPP-4253)

### PR DESCRIPTION
**Pull request by d4rken**
We no longer leave the screen if we cancel the cancel dialog :smile:.

* Change dialog cancel behavior. Stay on screen.
* Switch dialog button. The "positive" action is always in reference to the dialog's question. So positive here means to not warn others, while "warn others" is the negative action that cancels the dialog.
* Also removed some unused codes and did a little code cleanup.

## Testing
* Basically what is described in Jira: Positive TAN, press cancel on the test result screen, cancel the dialog.

### CODE REVIEW

**Review by ralfgehrer — APPROVED** [review]
Tested on Pixel (Android 10)

**Review by Oliver-Zimmerman — APPROVED** [review]
Tested on Pixel 5, works as expected. Code looks good to me.
